src/pages/ReportParticipantsPage.tsx

```diff
@@ -1,75 +1,82 @@
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetails, Report} from '@src/types/onyx';
+import type { ValueOf } from 'type-fest';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
 import withReportOrNotFound from './home/report/withReportOrNotFound';
-import personalDetailsPropType from './personalDetailsPropType';
-import reportPropTypes from './reportPropTypes';
 
-const propTypes = {
-    /* Onyx Props */
-
-    /** The personal details of the person who is logged in */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** The active report */
-    report: reportPropTypes.isRequired,
-
-    /** Route params */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** Report ID passed via route r/:reportID/participants */
-            reportID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    ...withLocalizePropTypes,
+type ReportParticipantsPageOnyxProps = {
+    personalDetails: OnyxCollection<PersonalDetails>;
 };
 
-const defaultProps = {
-    personalDetails: {},
-};
+type ReportParticipantsPageProps = ReportParticipantsPageOnyxProps & WithReportOrNotFoundProps;
+
+type ParticipantData = {
+    alternateText: string;
+    displayName: string;
+    accountID: number;
+    icons: Array<{
+        id: number;
+        source: UserUtils.AvatarSource;
+        name: string;
+        type: ValueOf<typeof CONST.ICON_TYPE_AVATAR>;
+    }>;
+    keyForList: string;
+    login: string;
+    text: string;
+    tooltipText: string;
+    participantsList: Array<{
+        accountID: number;
+        displayName: string;
+    }>;            
+}
 
 /**
  * Returns all the participants in the active report
  *
- * @param {Object} report The active report object
- * @param {Object} personalDetails The personal details of the users
- * @param {Object} translate The localize
- * @return {Array}
+ * @param report The active report object
+ * @param personalDetails The personal details of the users
+ * @param translate The localize
  */
-const getAllParticipants = (report, personalDetails, translate) =>
-    _.chain(ReportUtils.getVisibleMemberIDs(report))
+const getAllParticipants = (
+    report: OnyxEntry<Report>,
+    personalDetails: OnyxCollection<PersonalDetails>,
+    translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: Localize.PhraseParameters<Localize.Phrase<TKey>>) => string,
+): ParticipantData[] =>
+    ReportUtils.getVisibleMemberIDs(report)
         .map((accountID, index) => {
-            const userPersonalDetail = lodashGet(personalDetails, accountID, {displayName: personalDetails.displayName || translate('common.hidden'), avatar: ''});
-            const userLogin = LocalePhoneNumber.formatPhoneNumber(userPersonalDetail.login || '') || translate('common.hidden');
+            const userPersonalDetail = personalDetails?.[accountID];
```

Comment:
I think this is little bit different from the original

Reply:
Yes, it is but it works the same 😄 The old solution assigned a record that was found or this weird object (`{displayName: personalDetails.displayName || translate('common.hidden'), avatar: ''}`) which caused a lot of confusion in TS. Instead of creating that weird object I am just adding a fallback for `avatar` and a fallback for `displayName` is added by `getDisplayNameOrDefault` on **line 23**. Do you think we can keep it that way?